tasks/Base.cpp

```diff
@@ -241,33 +240,46 @@ bool Base::addCommandInput(std::string const & name, double timeout){
     return true;
 }
 
-bool Base::merge(bool const expected[], base::Vector3d const& current, base::Vector3d& merged){
-    for(int i = 0; i < 3; i++){
-        if(base::isUnset(current(i))){
-            continue;
-        }else if(_safe_mode.get() && !expected[i]){
-            if(state() != INPUT_UNEXPECTED){
-                error(INPUT_UNEXPECTED);
-            }
-            return false;
-        }else if(!base::isUnset(merged(i))){
-            //Ther is a value in the merged value and the value is set on this port.
-            //This is an error!
-            if(state() != INPUT_COLLIDING){
-                error(INPUT_COLLIDING);
-            }
-            return false; 
-        }else{
-            //No value of this type in the merged value and the value is set on this
-            //port. So write the Value from this Port in the merged value.
-            merged(i) = current(i);
-        }
+Base::States Base::merge(auv_control::ExpectedInputs const& expected, base::LinearAngular6DCommand const& current, base::LinearAngular6DCommand &merged)
+{
+    for(int i = 0; i < 3; i++)
+    {
+        if(_safe_mode.get() && !expected.linear[i] && !base::isUnset(current.linear[i]))
+            return INPUT_UNEXPECTED;
+        if(_safe_mode.get() && !expected.angular[i] && !base::isUnset(current.angular[i]))
+            return INPUT_UNEXPECTED;
+
+        //There is a value in the merged value and the value is set on this port.
+        //This is an error!
+        if(!base::isUnset(merged.linear[i]) && !base::isUnset(current.linear[i]))
+            return INPUT_COLLIDING;
+        if(!base::isUnset(merged.angular[i]) && !base::isUnset(current.angular[i]))
+            return INPUT_COLLIDING;
+
+        //No value of this type in the merged value and the value is set on this
+        //port. So write the Value from this Port in the merged value.
+        if(base::isUnset(merged.linear[i]))
+            merged.linear[i] = current.linear[i];
+        if(base::isUnset(merged.angular[i]))
+            merged.angular[i] = current.angular[i];
     }
-    return true;
+    return CONTROLLING;
 }
 
+std::vector<Base::InputPortInfo*> Base::checkConnectedPorts(std::vector<InputPortInfo> &in_ports) const
```

Comment:
I know it's not a hard requirement, but I would prefer that you avoid allocating on the heap during the updateHook. Keep an attribute in the class of type `std::vector<Base::InputPortInfo*>` that would be e.g. `connected_input_ports` and reserve() its size once and for all. Then, pass it as argument to this method.